spec/models/invenio_rdm_record_converter_spec.rb

```diff
@@ -302,6 +307,135 @@
+        ).to eql(expected_communities)
+      end
+    end
+
+    context "with two collections" do
+      before do
+        make_collection(user, title: "Collection", id: "collection-1",
+                        member_ids: [generic_file.id])
+      end
+
+      let(:expected_communities) {
+        [
+          [{"title": "Community", "id": "community-1"}.with_indifferent_access],
+          [{"title": "Collection", "id": "collection-1"}.with_indifferent_access]
+        ]
+      }
+
+      let(:collection_store) {
+        {"community-1": [
+          [{"title": "Community", "id": "community-1"}],
+          [{"title": "Collection", "id": "collection-1"}]
+        ]}.with_indifferent_access
+      }
+      let(:converted_record_with_two_collections) {
+        described_class.new(generic_file, collection_store)
+      }
+
+      it "adds data" do
+        expect(
+          converted_record_with_two_collections.send(:list_collections)[0]
+        ).to eq(expected_communities)
+      end
+    end
+
+    context "with parent collection" do
+      before do
+        make_collection(user, title: "Parent", id: "parent-1",
+                        member_ids: ["community-1"])
+      end
+
+      let(:expected_communities) {
+        [
+          [
+            {"title": "Parent", "id": "parent-1"}.with_indifferent_access,
+            {"title": "Community", "id": "community-1"}.with_indifferent_access
+          ]
+        ]
+      }
+
+      let(:collection_store) {
+        {"community-1": [
+          [
+            {"title": "Parent", "id": "parent-1"},
+            {"title": "Community", "id": "community-1"}
+          ]
+        ]}.with_indifferent_access
+      }
+      let(:converted_record_with_parent_collection) {
+        described_class.new(generic_file, collection_store)
+      }
+
+      it "adds data" do
+        expect(
+          converted_record_with_parent_collection.send(:list_collections)[0]
+        ).to eq(expected_communities)
+      end
+    end
+
+    context "with multiple parents collection" do
+      before do
+        make_collection(user, title: "Mom", id: "parent-1",
+                        member_ids: ["community-1"])
+        make_collection(user, title: "Dad", id: "parent-2",
+                        member_ids: ["community-1"])
+      end
+
+      let(:expected_communities) {
+        [
+          [{"title": "Mom", "id": "parent-1"}.with_indifferent_access,
+           {"title": "Community", "id": "community-1"}.with_indifferent_access],
+          [{"title": "Dad", "id": "parent-2"}.with_indifferent_access,
+           {"title": "Community", "id": "community-1"}.with_indifferent_access]
+        ]
+      }
+
+      let(:collection_store) {
+        {"community-1": [
+          [{"title": "Mom", "id": "parent-1"},
+           {"title": "Community", "id": "community-1"}],
+          [{"title": "Dad", "id": "parent-2"},
+           {"title": "Community", "id": "community-1"}]
+        ]}.with_indifferent_access
```

Comment:
Glad you were able to figure that out. I'm starting to wonder whether it makes more sense to just use hash rockets where you need to use characters incompatible with symbols in a hash or if there is some other solution. Something to look into some other time I guess.

Reply:
Yea, but as we both agreed on, the test should not have done any transforming. It should have tested this as it is, and everything would have been alright. It's not our fault, but geeze it was a headache to realize what was happening.